clang/test/SemaCXX/linkage3.cpp

```diff
@@ -0,0 +1,5 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -pedantic-errors %s
+
```

Comment:
Also add a test for this (at the top level w/o any linkage-spec-decl) if we don’t already have one:
```
int main;
```
which is also invalid; probably in a separate file so it doesn’t complain about redeclarations